src/puppet.js

```diff
@@ -842,7 +795,9 @@
     if(this.queue.pending && this.queue.pending.length && this.queue.pending.length > this.retransmissionThreshold) {
       // remote counterpart probably failed to receive one of earlier messages, because it has been receiving
       // (but not acknowledging messages for some time
-      this.queue.pending.forEach(this._sendPatches.bind(this));
+      this.queue.pending.forEach((function (p) {
+        sendPatches(this, p);
+      }).bind(this));
```

Comment:
If you are doing closure anyway, maybe we could get rid of `.bind` overhead?

I would either do
```js
this.queue.pending.forEach(function (p) {
    sendPatches(that, p);
});
```
or
```js
this.queue.pending.forEach(sendPatches.bind(null,this));
```